include/singa/worker.h

```diff
@@ -62,6 +62,12 @@ class Worker {
    * @return a pointer to the instance of the Worker subclass.
    */
   static Worker* Create(const AlgProto& conf);
+  /**
+   * Create an instance of the subclass of Worker
+   * with the specified device type.
+   */
+  static Worker* Create(const AlgProto& conf, int devid);
```

Comment:
can we bind the work to the device after creating the threads?
if so, then we do not need this API. 
instead, we can assign worker thread a device ID (gpu or cpu) through Context.
workers can then query Context to know its device IDs.